app/commands/put_content_with_links.rb

```diff
@@ -77,10 +77,23 @@ def content_item_attributes
     end
 
     def create_or_update_links!
-      LinkSet.create_or_replace(content_id: content_id, links: content_item[:links]) do |link_set|
+      LinkSet.create_or_replace(content_id: content_id) do |link_set|
         version = Version.find_or_initialize_by(target: link_set)
         version.increment
         version.save! if link_set.valid?
+
```

Comment:
If I'm understanding it all correctly, everything called from a controller will be called within `EventLogger.log_command`, which is wrapped in a transaction, for this reason.

https://github.com/alphagov/publishing-api/blob/7ecc62dbbdd1c4d47c193bc93c2602ce9e0f22d9/app/event_logger.rb#L7